### DISCUSSION TITLE
Web Version

### DISCUSSION THREAD

**Discussion by shahram7** [flxzt/rnote#1176] [Development]
Hi everyone,

Is there already a plan to develop a self-hosted web version of Rnote?

Best regards, Shahram

**Comment by Doublonmousse** [comment]
See #1082.
Since my other message the stroke modeler has been rewritten so that's at least one less problematic dependency for WebAssembly. The point about the rendering part still stands though. The renderer is tied to gtk, hence it won't work on the web.

It's possible we will change to another renderer in the future (wgpu based) which would be more portable (including the possibility of running it on the web with WebGPU and maybe mobile as well opening the way for a web version).

But it's not clear today how to do that (if we keep all of the ui code we don't know how to use wgpu, and if we don't and use another tool, the renderer would work but we won't have pen support so we don't want to do that today). 

Hence, maybe one day, but it definitely won't happen overnight
